system-helper/flatpak-system-helper.c

```diff
@@ -1151,6 +1302,355 @@ handle_run_triggers (FlatpakSystemHelper   *object,
   return TRUE;
 }
 
+static gboolean
+get_connection_uid (GDBusMethodInvocation *invocation, uid_t *out_uid, GError **error)
+{
+  GDBusConnection *connection = g_dbus_method_invocation_get_connection (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  g_autoptr(GVariant) dict = NULL;
+  g_autoptr(GVariant) credentials = NULL;
+
+  credentials = g_dbus_connection_call_sync (connection,
+                                             "org.freedesktop.DBus",
+                                             "/org/freedesktop/DBus",
+                                             "org.freedesktop.DBus",
+                                             "GetConnectionCredentials",
+                                             g_variant_new ("(s)", sender),
+                                             G_VARIANT_TYPE ("(a{sv})"), G_DBUS_CALL_FLAGS_NONE,
+                                             G_MAXINT, NULL, error);
+  if (credentials == NULL)
+    return FALSE;
+
+  dict = g_variant_get_child_value (credentials, 0);
+
+   if (!g_variant_lookup (dict, "UnixUserID", "u", out_uid))
+    {
+      g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                   "Failed to query UnixUserID for the bus name: %s", sender);
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static gboolean
+check_revokefs_requirements (GError **error)
+{
+  g_autofree gchar *revokefs_fuse_path = NULL;
+  struct passwd *passwd;
+
+  revokefs_fuse_path = g_find_program_in_path ("revokefs-fuse");
+  if (revokefs_fuse_path == NULL)
+    {
+      g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_NOT_SUPPORTED,
+                   "Can't find revoke-fuse on the system.");
+      return FALSE;
+    }
+
+  passwd = getpwnam ("flatpak");
+  if (passwd == NULL)
+    {
+      glnx_throw_errno_prefix (error, "User flatpak does not exist");
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static void
+revokefs_fuse_backend_child_setup (gpointer user_data)
+{
+  struct passwd *passwd = getpwnam ("flatpak");
+  setgid (passwd->pw_gid);
```

Comment:
You should probably `g_error` or otherwise abort if `setgid` or `setuid` fails. (This will only kill the subprocess, not the parent.)